### ISSUE TITLE
Make the module "Context Aware"

### ISSUE THREAD

**Issue by alba-ado** [pokusew/node-pcsclite#43]
Hello,

The nfc-pcsc library causes trouble with the newer versions of the electron ( After v16) because this module is not context aware. However, we solved this issue and confirmed to be working with nfc-pcsc in santigimeno's node-pcsclite repo as follows;

https://github.com/santigimeno/node-pcsclite/pull/96/commits/64952fb868feae7b93c0895a96e84aad0a7945f9

By just adding a few lines, we can make the module context aware. I tried this by myself on the nfc-pcsc library, and can confirm that it works without a problem (in the latest version of electron, which was 17 at the time of writing this). Can you make this change to the addon.cpp file and push this repo into the npm, so we can use the nfc-pcsc library with the new versions of the electron without changing the source files? (My understanding is that nfc-pcsc requires this repo, or am I wrong?)

Help is really appreciated. Thanks in advance.

**Comment by bastianandre-myenso** [comment]
Since there was no update by the maintainer so far I ended up doing the following:

1. Fork this repository
2. Apply the necessary changes to the fork
3. Install the forked package in your application

The package.json then looks like
```json
"dependencies": {
    "@pokusew/pcsclite": "github:<your-account>/node-pcsclite"
}
```

To use with electron v28+ I suggest also updating the nan peer dependency to v2.19 as suggested [here](/santigimeno/node-pcsclite/pull/102) from this [issue](santigimeno/node-pcsclite/issues/101).

**Comment by alba-ado** [comment]
Yes, I ended up doing something similar. But I wanted to notify the maintainers of this library to update the code as well so everyone can use it.

**Comment by martinpaljak** [comment]
Maybe it makes sense to try to make a new "official community managed fork" and while at it, drop the "lite" from the name, `node-pcsc` ? Maybe @LudovicRousseau has ideas or proposals?

**Comment by LudovicRousseau** [comment]
I have nothing to add.
I do not use node.js, except for my article https://blog.apdu.fr/posts/2014/09/pcsc-sample-in-javascript-nodejs/

I agree with @martinpaljak that it is strange to use the name "pcsclite" if it is not specific to pcsc-lite but can work with any PC/SC (or WinSCard) library.

**Comment by alba-ado** [comment]
> I have nothing to add. I do not use node.js, except for my article https://blog.apdu.fr/posts/2014/09/pcsc-sample-in-javascript-nodejs/
> 
> I agree with @martinpaljak that it is strange to use the name "pcsclite" if it is not specific to pcsc-lite but can work with any PC/SC (or WinSCard) library.

I think it comes from the Linux library name libpcsclite, that might be the naming reason

**Comment by alba-ado** [comment]
> Maybe it makes sense to try to make a new "official community managed fork" and while at it, drop the "lite" from the name, `node-pcsc` ? Maybe @LudovicRousseau has ideas or proposals?

I agree that we can make a community fork, but who will be in charge of this and wants to take responsibility? 😄
